lang/py3/avro/tests/gen_interop_data.py

```diff
@@ -46,11 +47,16 @@
 }
 
 
-if __name__ == "__main__":
-  interop_schema = schema.Parse(open(sys.argv[1], 'r').read())
-  writer = open(sys.argv[2], 'wb')
+def generate(schema_file, output_path):
+  interop_schema = schema.Parse(open(schema_file, 'r').read())
   datum_writer = io.DatumWriter()
-  # NB: not using compression
-  dfw = datafile.DataFileWriter(writer, datum_writer, interop_schema)
-  dfw.append(DATUM)
-  dfw.close()
+  for codec in datafile.VALID_CODECS:
+    filename = "py3{}.avro".format('' if codec == 'null' else '_' + codec)
+    writer = open(os.path.join(output_path, filename), 'wb')
```

Comment:
Please use a context manager (`with` statement) to manage open files. Also consider [Pathlib](https://docs.python.org/3/library/pathlib.html) here, as it may make the syntax a little more straightforward.

Reply:
Good catch, Thanks! Addressed the comments.